cryptoki/src/context/mod.rs

```diff
@@ -94,21 +94,37 @@ impl Pkcs11 {
         unsafe {
             let pkcs11_lib =
                 cryptoki_sys::Pkcs11::new(filename.as_ref()).map_err(Error::LibraryLoading)?;
-            let mut list = mem::MaybeUninit::uninit();
+            Self::_new(pkcs11_lib)
+        }
+    }
+
+    /// Instantiate a new context from current executable, the PKCS11 implementation is contained in the current executable
+    pub fn new_from_self() -> Result<Self> {
+        unsafe {
+            #[cfg(not(windows))]
+            let this_lib = libloading::os::unix::Library::this();
+            #[cfg(windows)]
+            let this_lib = libloading::os::windows::Library::this()?;
+            let pkcs11_lib = cryptoki_sys::Pkcs11::from_library(this_lib)?;
+            Self::_new(pkcs11_lib)
+        }
+    }
 
-            Rv::from(pkcs11_lib.C_GetFunctionList(list.as_mut_ptr()))
-                .into_result(Function::GetFunctionList)?;
+    unsafe fn _new(pkcs11_lib: cryptoki_sys::Pkcs11) -> Result<Self> {
```

Comment:
Is it idiomatic in Rust to have function names starting with `_` like this? Wondering if it could be a normal name instead, since the function is not public. But no big deal! Can be done later :)

Reply:
No preference on my side, it's just something we commonly do at my work :)

Reply:
Ok let's go with it! If someone has strong opinions they can change in the future :)